target/linux/ath79/image/generic.mk

```diff
@@ -50,6 +50,30 @@ define Build/cybertan-trx
 	-rm $@-empty.bin
 endef
 
+define Build/belkin_f9j1108-v2-factory
+       cat $@.uImage > $@.new
+       cat $@.rootfs >> $@.new
+       mv $@.new $@
+       rm -rf $@.*
+endef
+
+define Build/edimax-headers
+	$(eval device_name=$(word 1,$(1)))
+	$(eval edimax_magic=$(word 2,$(1)))
+	$(eval edimax_model=$(word 3,$(1)))
+
+	$(STAGING_DIR_HOST)/bin/edimax_fw_header -M $(edimax_magic) -m $(edimax_model)\
+		-v $(VERSION_DIST)$(firstword $(subst +, , $(firstword $(subst -, ,$(REVISION))))) \
+		-n "uImage" \
+		-i $(KDIR)/loader-$(device_name).uImage \
+		-o $@.uImage;
```

Comment:
If you merge this with the function above, you essentially don't need a separate $@.uImage which you then just copy again into $@.new; I'd just write to $@.new directly here. Or you stay with $@.uImage here and just concatenate $@.uImage and $@.rootfs together directly into $@ without the bypass via $@.new ...

Reply:
I think the issue was around concurrency, where the files didn't exist at the right time, I will combine the functions as suggested, and test for concurrency issues.